Refactor AXFR code

nemanja-zivkovic-atomia opened the pull request:
Refactored AXFR code to not use domainmetadata_change table anymore.
Added new trigger and new trigger function on domainmetadata.
Added TSIG key handling trough change and slavezone_change tables.
Added GetChangedZonesBatchWithTSIG and GetChangedSlaveZonesWithTSIG.
Added a new return type changedzonewithtsig.
Other minor changes.

Partially resolves PROD-2760 (1/2).

nemanja-zivkovic-atomia committed:
Refactor AXFR code

Refactored AXFR code to not use domainmetadata_change table anymore.
Added new trigger and new trigger function on domainmetadata.
Added TSIG key handling trough change and slavezone_change tables.
Added GetChangedZonesBatchWithTSIG and GetChangedSlaveZonesWithTSIG.
Added a new return type changedzonewithtsig.
Other minor changes.

Partially resolves PROD-2760 (1/2).

nemanja-zivkovic-atomia committed:
Improve GetChangedZonesBatchWithTSIG SQL performance

Improved GetChangedZonesBatchWithTSIG SQL performance by 18%.

Amends PROD-2760.

jovana-stamenkovic-atomia commented on server/schema/getchangedslavezoneswithtsig.sql:
```diff
@@ -0,0 +1,24 @@
+			INNER JOIN nameserver ON nameserver_id = nameserver.id
+			INNER JOIN slavezone ON zone = slavezone.name LEFT JOIN domainmetadata ON slavezone.id = domainmetadata.domain_id AND domainmetadata.kind = 'slave'
+			WHERE nameserver.name = nameservername AND status = 'PENDING' 
+			ORDER BY changetime ASC, slavezone_change.id ASC
```
After calling DeleteSlaveZone, the zone is removed from the slave zone table and there is an entry in the 'slavezone_change' table. However, this function doesn't return the change. This is probably because of ```INNER JOIN slavezone ON zone = slavezone.name```, the slave zone doesn't exist.

jovana-stamenkovic-atomia commented on server/schema/getchangedzonesbatchwithtsig.sql:
```diff
@@ -0,0 +1,26 @@
+			WHERE nameserver.name = nameservername AND status = 'PENDING'
+			GROUP BY zone
+			LIMIT changelimit) AS res
+			INNER JOIN zone ON res.zone = zone.name LEFT JOIN domainmetadata ON zone.id = domainmetadata.domain_id AND domainmetadata.kind = 'master'
```
Check the same as above.

nemanja-zivkovic-atomia committed:
Fix join issue with the new GetChanged functions

Fixed join issue with GetChangedZonesBatchWithTSIG and
GetChangedSlaveZonesWithTSIG functions where they would not fetch
rows if zone was deleted.

Amends PROD-2760.

jovana-stamenkovic-atomia commented on server/schema/assigntsigkey.sql:
```diff
@@ -1,12 +1,12 @@
 CREATE OR REPLACE FUNCTION AssignTSIGKey(
 	domain varchar,
```
@nemanja-zivkovic-atomia It is possible to assign the tsig key that doesn't exist from atomiadns. This is not a valid situation from HCP. I am not sure if this should be fixed?

nemanja-zivkovic-atomia replied:
It is possible, only trough AtomiaDNS. It is not possible trough HCP.

@sasa-mladenovic-troxo I don't remember if we discussed this, but should we prevent this situation? We can do so by adding a foreign key in domainmetadata on tsigkey table.

sasa-mladenovic-troxo replied:
This seems to be allowed on PowerDNS as well. I'm fine to leave it like this for now.

jovana995 committed:
Refactor bind syncer slave zones

Refactored bind syncer slave zones.

Amends PROD-2760.

jovana995 committed:
Rename the property for tsig keys

Renamed the property for tsig keys

Amends PROD-2760.

nemanja-zivkovic-atomia committed:
Bump version to 1.1.57

nemanja-zivkovic-atomia committed:
Merge branch 'develop' of ssh://github.com/atomia/atomiadns into develop

nemanja-zivkovic-atomia committed:
Merge branch 'develop' of ssh://github.com/atomia/atomiadns into develop

nemanja-zivkovic-atomia committed:
Fix an issue due to domainmetadata_change not existing

This issue can happen because we are still doing
reload_updated_domainmetadata (for backward compatibility)
and because we are not creating the table during clean install
(as we shouldn't).
reload_updated_domainmetadata calls GetChangedDomainIDs which reads
from domainmetadata_change which may not exist.

nemanja-zivkovic-atomia committed:
Merge branch 'develop' of ssh://github.com/atomia/atomiadns into develop

nemanja-zivkovic-atomia committed:
Merge branch 'develop' into PROD-2760

# Conflicts:
#	server/debian/atomiadns-database.postinst
#	server/generate_wsdl.sh
#	server/schema/ddl.sql
#	server/schema/unassigntsigkey.sql
#	server/wsdl-atomiadns.wsdl
#	server/wsdl-type-schema.xsd

nemanja-zivkovic-atomia committed:
Update AXFR refactor to be backwards compatible

Update AXFR refactor to be backwards compatible and compatible with
a minor release.
Reverted AssignTSIGKey function to include kind parameter to enable
backwards compatibility.
Changed "use_tsig_keys" config to "disable_tsig_keys" in PowerDNS
syncer and inverted the condition so that it doesn't disable
TSIG keys by default.
Changed "use_tsig_keys_for_slave_zones" to "disable_tsig_keys" in
bind syncer to be consistent with PowerDNS syncer and so that it
doesn't disable TSIG keys by default.
Removed use_tsig_keys_for_slave_zones function as it's no longer needed.
Improved exception handling of assign_tsig_key syncer function.
Removed "use_tsig_keys" from the config file.

Amends PROD-2760

nemanja-zivkovic-atomia committed:
Split fetching TSIG keys from fetching changed zones

Split functions GetChangedZonesBatchWithTSIG and
GetChangedSlaveZonesWithTSIG to fetch zones/slavezones separate
from TSIG keys.

Amends PROD-2760

nemanja-zivkovic-atomia committed:
Split fetching TSIG keys from fetching changed zones in bindsync

Split functions GetChangedZonesBatchWithTSIG and
GetChangedSlaveZonesWithTSIG to fetch zones/slavezones separate
from TSIG keys in bindsync.

Amends PROD-2760

nemanja-zivkovic-atomia committed:
Rename functions for handling tsig key assignment fetch

Renamed functions for handling tsig key assignment fetch:
atomiaTSIGKeyList -> atomiaTSIGKeyAssignmentList
atomiaTSIGKeyItem -> atomiaTSIGKeyAssignmentItem
